### PR TITLE
[JENKINS-42000] Communication errors in organization scanning should not delete all everything

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-42000](https://issues.jenkins-ci.org/browse/JENKINS-42000)

@reviewbybees

### CODE REVIEW

**Comment by stephenc** [comment]
Downstream of https://github.com/jenkinsci/scm-api-plugin/pull/29

**Comment by KostyaSha** [comment]
So simple fix requires changing 3 plugins???

**Comment by stephenc** [comment]
@KostyaSha no, I technically didn't need to fix GHBS or BBBS but it is better to keep them on up to date dependencies, so for them I am fixing the compiler error proactively

**Comment by stephenc** [comment]
@KostyaSha and when you separate the API from the implementation, a change to the API contract will always have downstream impact. 

In this case the only place where it is *runtime* required to fix the plugin is in the `OrganizationFolder`. In fact I could probably have hacked by either throwing a `RuntimeException` and then catching it again or by using an `AtomicReference` to stash the first exception... but other consumers of the SCM API would then have to repeat the "hack"

File: pom.xml [hunk 1]
```diff
@@ -66,7 +66,8 @@
 
   <properties>
     <jenkins.version>1.642.3</jenkins.version>
-    <scm-api.version>2.0.3</scm-api.version>
+    <scm-api.version>2.0.4-20170214.145531-1</scm-api.version>
+    <hpi-plugin.version>1.121</hpi-plugin.version> <!-- TODO remove with the timestamp -SNAPSHOT -->
```
Comment (jglick) [inline pom.xml]: Just update the parent to 2.22.

File: pom.xml [hunk 2]
```diff
@@ -66,7 +66,8 @@
 
   <properties>
     <jenkins.version>1.642.3</jenkins.version>
-    <scm-api.version>2.0.3</scm-api.version>
+    <scm-api.version>2.0.4-20170214.145531-1</scm-api.version>
```
Comment (jglick) [inline pom.xml]: recommend linking to the upstream PR

File: src/test/java/integration/EventsTest.java [hunk 3]
```diff
@@ -1204,6 +1204,111 @@ public void given_orgFolder_when_someReposMatch_then_scanCreatesMatchingProjects
     }
 
     @Test
```
Comment (jglick) [inline src/test/java/integration/EventsTest.java]: `@Issue`?

**Pull request merged**